backend/compact-connect/lambdas/purchases/handlers/privileges.py

```diff
@@ -42,3 +61,133 @@ def get_purchase_privilege_options(event: dict, context: LambdaContext):  # noqa
+    if len(selected_jurisdictions) != len(selected_jurisdictions_postal_abbreviations):
+        # this could only happen if the jurisdiction configuration was not uploaded or was deleted somehow
+        logger.error(
+            'Jurisdiction configuration missing. Requested jurisdiction not found.',
+            existing_jurisdiction_configuration=[
+                selected_jurisdiction.postal_abbreviation for selected_jurisdiction in selected_jurisdictions
+            ],
+            selected_jurisdictions_postal_abbreviations=selected_jurisdictions_postal_abbreviations,
+        )
+        raise CCInvalidRequestException('Invalid jurisdiction postal code')
+
+    # get the user's profile information to determine if they are active military
+    provider_id = _get_caller_provider_id_custom_attribute(event)
+    user_provider_data = config.data_client.get_provider(compact=compact_name, provider_id=provider_id)
+    provider_record = next((record for record in user_provider_data['items'] if record['type'] == 'provider'), None)
+    license_record = _find_best_license(
+        [record for record in user_provider_data['items'] if record['type'] == 'license']
+    )
+    # this should never happen, but we check just in case
+    if provider_record is None:
+        raise CCNotFoundException('Provider not found')
+    if license_record is None:
+        raise CCNotFoundException('License record not found for this user')
+
+    license_expiration_date: date = license_record['dateOfExpiration']
+    user_active_military = bool(provider_record.get('militaryWaiver', False))
+
+    purchase_client = PurchaseClient()
+    transaction_response = None
+    try:
+        transaction_response = purchase_client.process_charge_for_licensee_privileges(
+            order_information=body['orderInformation'],
+            compact_configuration=compact,
+            selected_jurisdictions=selected_jurisdictions,
+            user_active_military=user_active_military,
+        )
+
+        # transaction was successful, now we create privilege records for the selected jurisdictions
+        config.data_client.create_provider_privileges(
+            compact_name=compact_name,
+            provider_id=provider_id,
+            jurisdiction_postal_abbreviations=selected_jurisdictions_postal_abbreviations,
+            license_expiration_date=license_expiration_date,
+            compact_transaction_id=transaction_response['transactionId'],
+        )
+
+        return transaction_response
+
+    except CCFailedTransactionException as e:
+        logger.warning(f'Failed transaction: {e}.')
+        raise CCInvalidRequestException(f'Error: {e.message}') from e
+    except CCAwsServiceException as e:
+        logger.error(f'Error creating privilege records: {e.message}. Voiding transaction.')
+        if transaction_response:
+            # void the transaction if it was successful
+            purchase_client.void_privilege_purchase_transaction(
```

Comment:
Note to self- This seems very important to test thoroughly.

Reply:
FWIW - I did verify this locally and confirmed it voids a transaction given a valid transaction id.